test/components/util/util.go

```diff
@@ -36,6 +36,16 @@ import (
 	"k8s.io/client-go/transport/spdy"
 )
 
+const (
+	// RetryInterval is the default retry interval for tests using retry strategy, so they
+	// don't have to specify their own local intervals.
+	RetryInterval = time.Second * 5
```

Comment:
This prepares a ground for a better code, by adding the const and then actually using them. Refactoring is out of scope of this PR. And I want to avoid adding yet another duplication.

This way, newly added code should use those const, meaning once we actually do the refactoring, we have less work to do.

Reply:
I'm not against adding the constants, I'm against adding them in a separate file because we *might* need to do that at some unknown point in the future.

Would you consider doing the change when it is actually needed rather than based on a theoretical predication of the future? The price we are paying by doing this right now is that we expand our reliance on "util" packages. If we put any code piece which we *may* be able to reuse in the future in a "util" package, we'd gradually be moving most of our logic there.

If you do the optimization once you encounter a *concrete* problem (e.g. "oh, I need to duplicate this code to make it work"), you will have a better idea where to put this code based on the concrete case you have.

Reply:
I would have to duplicate this code already:
```
$ git grep 'time.Minute\*5' test/
test/components/cert-manager/cert-manager_test.go:                      testutil.WaitForDeployment(t, client, namespace, test.deployment, time.Second*5, time.Minute*5)
test/components/contour/contour_test.go:        testutil.WaitForDaemonSet(t, client, namespace, daemonset, time.Second*5, time.Minute*5)
test/components/contour/contour_test.go:        testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/coredns/coredns_test.go:        testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/dex/dex_test.go:                testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/external-dns/external-dns_test.go:              testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/flatcar-linux-update-operator/fluo_test.go:     testutil.WaitForDaemonSet(t, client, namespace, daemonset, time.Second*5, time.Minute*5)
test/components/flatcar-linux-update-operator/fluo_test.go:     testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/gangway/gangway_test.go:                testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/metallb/metallb_test.go:                testutil.WaitForDaemonSet(t, client, namespace, daemonset, time.Second*5, time.Minute*5)
test/components/metallb/metallb_test.go:                testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/metrics-server/metrics_server_test.go:          testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/openebs-operator/openebs_operator_test.go:                      testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
test/components/rook/rook_test.go:              testutil.WaitForDeployment(t, client, namespace, deployment, time.Second*5, time.Minute*5)
```

This change will prevent exactly that.

Reply:
I understand you are bothered by "duplicating" the expression `time.Minute*5` inside `test/components`. I don't understand how the `test/components` package is related to this PR (which touches SSH keys on Packet). I also don't think that repeating `time.Minute*5` can be considered code duplication, *unless* we are assuming that all these tests must use the same value there. If the "duplication" in `test/components` is what we're discussing here, I'd say it's completely out of scope and move that part into a separate PR.

FWIW, **if** we decide that all tests should use the same `time.Minute*5` value, the way I would solve it is by defining the constant in `test/components` and avoid the `util` package. To be precise, I'd put the following code in `test/components/components.go` (or even `util.go` if you insist):

```go
package components

const (
    RetryInterval = time.Second * 5
)
```

For **this** PR I would not use a constant that's intended for using by components, because the code in this PR isn't related to components. In case you are arguing that we should use one global value for **all tests**, I would simply define the constant in the `test` package and import it into any "child" package under `test/`.

Reply:
In case I wasn't able to convince you with the above, please go ahead with whatever works for you. We need this change to get merged and I don't want to block it any further.

Reply:
FWIW I agree with @johananl here.

Reply:
I removed the constants.